apps/webapp/src/clients.ts

```diff
@@ -0,0 +1,34 @@
+import { createPromiseClient } from '@connectrpc/connect';
+import { ViewService } from '@buf/penumbra-zone_penumbra.connectrpc_es/penumbra/view/v1/view_connect';
+import { createChannelTransport } from '@penumbra-zone/transport';
+import { SimulationService } from '@buf/penumbra-zone_penumbra.connectrpc_es/penumbra/core/component/dex/v1/dex_connect';
+import { CustodyService } from '@buf/penumbra-zone_penumbra.connectrpc_es/penumbra/custody/v1/custody_connect';
+import { Query as IbcClientService } from '@buf/cosmos_ibc.connectrpc_es/ibc/core/client/v1/query_connect';
+import { jsonOptions } from '@penumbra-zone/types/src/json-options';
+
+import { penumbra } from '@penumbra-zone/types/penumbra-global';
+
+export const getPenumbraPort = async () => {
+  if (!window[penumbra]) throw Error('No Penumbra global (no provider installed)');
+
+  const providers = Array.from(Object.values(window[penumbra]));
```

Comment:
```suggestion
  const providers = Object.values(window[penumbra]);
```
(Right? Do we need `Array.from()` here?)